src/stan/services/pathfinder/multi.hpp

```diff
@@ -0,0 +1,210 @@
+ * @tparam SingleParamWriter Type inheriting from stan::callbacks::writer
+ * @param[in] model defining target log density and transforms (log $p$ in
+ * paper)
+ * @param[in] init ($pi_0$ in paper) var context for initialization. Random
+ * initial values will be generated for parameters user has not supplied.
+ * @param[in] random_seed seed for the random number generator
+ * @param[in] path path id to advance the pseudo random number generator
+ * @param[in] init_radius A non-negative value to initialize variables uniformly
+ * in (-init_radius, init_radius) if not defined in the initialization var
+ * context
+ * @param[in] history_size  Non-negative value for (J in paper) amount of
+ * history to keep for L-BFGS
+ * @param[in] init_alpha Non-negative value for line search step size for first
+ * iteration
+ * @param[in] tol_obj Non-negative value for convergence tolerance on absolute
+ * changes in objective function value
+ * @param[in] tol_rel_obj ($tau^{rel}$ in paper) Non-negative value for
+ * convergence tolerance on relative changes in objective function value
+ * @param[in] tol_grad Non-negative value for convergence tolerance on the norm
+ * of the gradient
+ * @param[in] tol_rel_grad Non-negative value for convergence tolerance on the
+ * relative norm of the gradient
+ * @param[in] tol_param Non-negative value for convergence tolerance changes in
+ * the L1 norm of parameter values
+ * @param[in] num_iterations (L in paper) Non-negative value for maximum number
+ * of LBFGS iterations
+ * @param[in] save_iterations indicates whether all the iterations should
+ *   be saved to the parameter_writer
+ * @param[in] refresh Output is written to the logger for each iteration modulo
+ * the refresh value
+ * @param[in,out] interrupt callback to be called every iteration
+ * @param[in] num_elbo_draws (K in paper) number of MC draws to evaluate ELBO
+ * @param[in] num_draws (M in paper) number of approximate posterior draws to
+ * return
+ * @param[in] num_multi_draws The number of draws to return from PSIS sampling
+ * @param[in] num_paths The number of single pathfinders to run.
+ * @param[in,out] logger Logger for messages
+ * @param[in,out] init_writers Writer callback for unconstrained inits
+ * @param[in,out] single_path_parameter_writer output for parameter values of
+ * the individual pathfinder runs.
+ * @param[in,out] single_path_diagnostic_writer output for diagnostics values of
+ * the individual pathfinder runs.
+ * @param[in,out] parameter_writer output for parameter values
+ * @param[in,out] diagnostic_writer output for diagnostics values,
+ * `error_codes::SOFTWARE` for failures
+ * @return error_codes::OK if successful
+ */
+template <class Model, typename InitContext, typename InitWriter,
+          typename DiagnosticWriter, typename ParamWriter,
+          typename SingleParamWriter, typename SingleDiagnosticWriter>
+inline int pathfinder_lbfgs_multi(
```

Comment:
order of arguments should match between single and multi as much as possible.
across servies, is there consistency in the way that args are ordered?  if so, we should conform.

Reply:
:+1 fixed